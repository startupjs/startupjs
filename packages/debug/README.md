# startupjs debug

> Debugging helpers

## Usage

For internal use only.

## License

MIT

(c) Decision Mapper - http://decisionmapper.com
