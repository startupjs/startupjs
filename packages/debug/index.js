export const DEBUG = {}

if (typeof window !== 'undefined') {
  if (!window.__startupjs__) window.__startupjs__ = {}
  window.__startupjs__.DEBUG = DEBUG
}

export function __increment (name) {
  if (!DEBUG[name]) DEBUG[name] = 0
  DEBUG[name] += 1
}

export function __decrement (name) {
  if (!DEBUG[name]) DEBUG[name] = 0
  DEBUG[name] -= 1
}

export function setDebugVar (key, value) {
  DEBUG[key] = value
}
