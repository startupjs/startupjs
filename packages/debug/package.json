{
  "name": "@startupjs/debug",
  "version": "0.55.0-alpha.1",
  "type": "module",
  "engines": {
    "node": ">= 14"
  },
  "description": "Debugging helpers",
  "main": "index.js",
  "publishConfig": {
    "access": "public"
  },
  "license": "MIT"
}
