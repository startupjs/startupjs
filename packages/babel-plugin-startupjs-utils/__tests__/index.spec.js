const pluginTester = require('babel-plugin-tester').default
const plugin = require('../index')
const { name: pluginName } = require('../package.json')

pluginTester({
  plugin,
  pluginName,
  snapshot: true,
  babelOptions: {
    plugins: ['@babel/plugin-syntax-jsx']
  },
  pluginOptions: {},
  tests: {
    'Use default cache value if nothing is set in options': /* js */`
      import cacheEnabled from '@startupjs/cache/enabled'
      import { __increment, __decrement } from '@startupjs/debug'

      export const CACHE_ACTIVE = { value: cacheEnabled }
    `
  }
})

pluginTester({
  plugin,
  pluginName,
  snapshot: true,
  babelOptions: {
    plugins: ['@babel/plugin-syntax-jsx']
  },
  pluginOptions: {
    observerCache: true
  },
  tests: {
    'Get cache value from options': /* js */`
      import cacheEnabled from '@startupjs/cache/enabled'
      import { __increment, __decrement } from '@startupjs/debug'

      export const CACHE_ACTIVE = { value: cacheEnabled }
    `
  }
})

pluginTester({
  plugin,
  pluginName,
  snapshot: true,
  babelOptions: {
    plugins: ['@babel/plugin-syntax-jsx']
  },
  pluginOptions: {},
  tests: {
    'Use default signals value if nothing is set in options': /* js */`
      import signalsEnabled from '@startupjs/signals/enabled.js'

      export const SIGNALS_ACTIVE = { value: signalsEnabled }
    `
  }
})

pluginTester({
  plugin,
  pluginName,
  snapshot: true,
  babelOptions: {
    plugins: ['@babel/plugin-syntax-jsx']
  },
  pluginOptions: {
    signals: true
  },
  tests: {
    'Get signals value from options': /* js */`
      import signalsEnabled from '@startupjs/signals/enabled.js'

      export const SIGNALS_ACTIVE = { value: signalsEnabled }
    `
  }
})
