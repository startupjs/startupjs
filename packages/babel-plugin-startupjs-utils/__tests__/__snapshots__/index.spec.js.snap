// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@startupjs/babel-plugin-startupjs-utils Get cache value from options: Get cache value from options 1`] = `

import cacheEnabled from '@startupjs/cache/enabled'
import { __increment, __decrement } from '@startupjs/debug'

export const CACHE_ACTIVE = { value: cacheEnabled }

      ↓ ↓ ↓ ↓ ↓ ↓

import { __increment, __decrement } from "@startupjs/debug";
const cacheEnabled = true;
export const CACHE_ACTIVE = {
  value: cacheEnabled,
};


`;

exports[`@startupjs/babel-plugin-startupjs-utils Get signals value from options: Get signals value from options 1`] = `

import signalsEnabled from '@startupjs/signals/enabled.js'

export const SIGNALS_ACTIVE = { value: signalsEnabled }

      ↓ ↓ ↓ ↓ ↓ ↓

const signalsEnabled = true;
export const SIGNALS_ACTIVE = {
  value: signalsEnabled,
};


`;

exports[`@startupjs/babel-plugin-startupjs-utils Use default cache value if nothing is set in options: Use default cache value if nothing is set in options 1`] = `

import cacheEnabled from '@startupjs/cache/enabled'
import { __increment, __decrement } from '@startupjs/debug'

export const CACHE_ACTIVE = { value: cacheEnabled }

      ↓ ↓ ↓ ↓ ↓ ↓

import { __increment, __decrement } from "@startupjs/debug";
const cacheEnabled = false;
export const CACHE_ACTIVE = {
  value: cacheEnabled,
};


`;

exports[`@startupjs/babel-plugin-startupjs-utils Use default signals value if nothing is set in options: Use default signals value if nothing is set in options 1`] = `

import signalsEnabled from '@startupjs/signals/enabled.js'

export const SIGNALS_ACTIVE = { value: signalsEnabled }

      ↓ ↓ ↓ ↓ ↓ ↓

const signalsEnabled = false;
export const SIGNALS_ACTIVE = {
  value: signalsEnabled,
};


`;
